### PR TITLE
Implementation of limit on purchase of specific items

### PR DESCRIPTION
**Pull request by agrawalarpit14**
In this PR:
- `limitPerUserProuductIds` denotes the Product Ids of the products which have a limit on purchase per user(5 units).
- `insecurity.appendUserId()` appends the true user id of the user to the request, for further use in routes.
- For new and existing API tests to pass, `User Id` is added to `authHeader`.

### CODE REVIEW

**Comment by agrawalarpit14** [comment]
Hi!! @bkimminich @J12934, please let me know your feedback.
Thank You😁😁✌🏻

File: lib/insecurity.js [hunk 1]
```diff
@@ -107,3 +107,11 @@ exports.roles = {
   accounting: 'accounting',
   admin: 'admin'
 }
+
+exports.appendUserId = () => {
```
Comment (J12934) [inline lib/insecurity.js:111]: This should probably done via the token map (also located in insecurity.js).
Unless I'm missing why this is done this way 😉

File: test/api/basketApiSpec.js [hunk 2]
```diff
@@ -4,7 +4,7 @@ const insecurity = require('../../lib/insecurity')
 const API_URL = 'http://localhost:3000/api'
 const REST_URL = 'http://localhost:3000/rest'
 
-const authHeader = { 'Authorization': 'Bearer ' + insecurity.authorize(), 'content-type': 'application/json' }
+const authHeader = { 'Authorization': 'Bearer ' + insecurity.authorize({ data: { id: 2 } }), 'content-type': 'application/json' }
```
Comment (J12934) [inline test/api/basketApiSpec.js]: Why was this changed?
Seems weird that this is the only change in the file.

**Review by J12934 — APPROVED** [review]
